src/main/java/com/google/step/YOUR_PROJECT_NAME_HERE/external/StackOverflowClient.java

```diff
@@ -0,0 +1,152 @@
+import java.util.regex.Pattern;
+import org.apache.http.HttpEntity;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.json.JSONObject;
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+import org.jsoup.nodes.Element;
+import org.jsoup.select.Elements;
+
+public final class StackOverflowClient {
+  private static final String SEARCH_URL_TEMPLATE =
+      "https://api.stackexchange.com/2.2/questions/%s?"
+          + "order=desc&sort=activity&site=stackoverflow";
+  private static final String QUESTION_URL_TEMPLATE =
+      "https://api.stackexchange.com/2.2/questions/%s/answers?"
+          + "order=desc&sort=votes&site=stackoverflow";
+  // This url specify filter to generate answer body.
+  private static final String ANSWER_URL_TEMPLATE =
+      "https://api.stackexchange.com/2.2/answers/%s?order"
+          + "=desc&sort=activity&site=stackoverflow&filter=!9_bDE(fI5";
+  private static final int ID_INDEX = 2;
+  private static final String ITEM_PARAMETER = "items";
+  private static final String TITLE_PARAMETER = "title";
+  private static final String BODY_PARAMETER = "body";
+  private static final String CODE_PARAMETER = "code";
+  private static final String ANSWER_ID_PARAMETER = "answer_id";
+  // Set 200 to be the maximum length of description for MVP.
+  private static final int DESCRIPTION_LENGTH_PARAMETER = 200;
+
+  /* This method build the desired card after getting each field. */
+  /* It will return a null card if no valid question is found */
+  public Card search(String url) {
+    String questionId;
+    try {
+      questionId = getQuestionId(url);
+    } catch (URISyntaxException e) {
+      // Return null card if no valid card available.
+      return null;
+    }
+    String answerId = getAnswerId(questionId);
+    if (answerId == null) {
+      return null;
+    }
+    String title = getTitle(questionId);
+    String answerBody = getAnswerBody(answerId);
+    String description = getDescription(answerBody);
+    String code = getCode(answerBody);
+    Card card = new Card(title, code, url, description);
+    return card;
+  }
+
+  /* Get the question id of passed in URL. */
+  private String getQuestionId(String url) throws URISyntaxException {
+    URI uri = new URI(url);
+    // Parse the URL to get the question id.
+    String[] segments = uri.getPath().split("/");
+    String questionId = segments[ID_INDEX];
+    if (!Pattern.matches("[0-9]+", questionId)) {
+      return null;
+    }
+    return questionId;
+  }
+
+  /* Get the question title using question id */
+  private String getTitle(String questionId) {
+    String searchUrl = String.format(SEARCH_URL_TEMPLATE, questionId);
+    String title = getResponse(searchUrl, TITLE_PARAMETER);
+    return title;
+  }
+
+  /* Get the most voted answer's id and store it in the card. */
```

Comment:
Since you're not storing the answerId in the card anymore, you can change this to just be `Return the ID of the highest voted answer.`

Likewise, modify the comments for your other `get` methods.